src/subscription/azext_subscription/custom.py

```diff
@@ -59,31 +63,47 @@ def _get_object_id(graph_client, subscription=None, spn=None, upn=None):
 
 def _object_id_args_helper(cli_ctx, object_id=None, spn=None, upn=None):
     if not object_id:
-        from azure.cli.core._profile import Profile
-        from azure.graphrbac import GraphRbacManagementClient
-
-        profile = Profile(cli_ctx=cli_ctx)
-        cred, _, tenant_id = profile.get_login_credentials(
-            resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-        graph_client = GraphRbacManagementClient(cred,
-                                                 tenant_id,
-                                                 base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+        from azure.cli.command_modules.role import graph_client_factory
+        graph_client = graph_client_factory(cli_ctx)
```

Comment:
The new `azure.cli.command_modules.role._msgrpah._graph_client.GraphClient` is very different from `azure.graphrbac.GraphRbacManagementClient`. See

- https://learn.microsoft.com/en-us/cli/azure/microsoft-graph-migration
- https://github.com/Azure/azure-cli/blob/dev/doc/microsoft_graph_client.md

A simple replacement will break things such as https://github.com/Azure/azure-cli/issues/30390.

https://github.com/Azure/azure-cli/pull/28105 introduced some centralized functions in `azure.cli.command_modules.role.util` for object ID resolution.